lib/test-utils/src/Test/QuickCheck/Extra.hs

```diff
@@ -51,3 +57,22 @@ shrinkInterleaved (a, shrinkA) (b, shrinkB) = interleave
     interleave (x : xs) (y : ys) = x : y : interleave xs ys
     interleave xs [] = xs
     interleave [] ys = ys
+
+--------------------------------------------------------------------------------
+-- Combinators
+--------------------------------------------------------------------------------
+
+-- | A combinator for generating larger arbitrary values, produced by
+--   generating many smaller arbitrary values and folding them together.
+--
+-- Applying this combinator to a type 'A' with an instance of 'Monoid' will
+-- generate larger values of 'A', based on simple monoidal concatenation of
+-- multiple smaller values.
+--
+newtype ManyFolded a = ManyFolded
```

Comment:
This is required because the available Arbitrary instance for TokenMap is something like this:

```
instance Arbitrary TokenMap where
    arbitrary = genTokenMapSmallRange
```

right?

I think I prefer Hedgehog's usage of functions to Quickcheck's typeclass instances, but that's not the question here.

Reply:
I found this confusing @jonathanknowles.

Adding in `Blind` didn't help (couldn't `Show` be defined with `show . mconcat`?

Why can't we just use the QuickCheck `size` parameter? See [Coding Standards - QuickCheck](https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#use-the-size-parameter).

Reply:
Thanks @rvl. I agree with you that using the `size` parameter would be a better solution here.

I've added a commit which does this here: 86f48565e9822e3cb1d76721236e621c152e72e5